Makefile

```diff
@@ -262,9 +275,9 @@ audit: sign audit-licenses
 
 publish: audit publish-bin publish-pip publish-src publish-jars
 
-all: clean test audit
+all: clean system-test test audit
```

Comment:
It looks like `system-test` which depends on `pip-release` compiles locally as does `test`.  From looking at the logs at test it looks like when it gets to `test` it eagerly compiles less (the test classes only).  If we move `test` before I think it will compile everything and when it gets to `pip-release` that part will be automatically skipped rather than the compilation process running partially in both targets.

See [here for test compilation build logs](https://travis-ci.org/apache/incubator-toree/builds/129268926#L1133)
See [here for pip release which does assembly which skips compiling test classes build logs](https://travis-ci.org/apache/incubator-toree/builds/129268926#L384)

I will reverse the order because that may make the compilation process better and intuitively it kind of makes more sense to detect unit test failures before getting to system tests.